controller/instance_manager_controller.go

```diff
@@ -1461,15 +1464,37 @@ func (imc *InstanceManagerController) createGenericManagerPodSpec(im *longhorn.I
 	return podSpec, nil
 }
 
+func getLivenessProbeCommand(dataEngine longhorn.DataEngineType) string {
+	var livenessProbes []string
+
+	ports := []int{
+		engineapi.InstanceManagerProcessManagerServiceDefaultPort,
+		engineapi.InstanceManagerProxyServiceDefaultPort,
+		engineapi.InstanceManagerDiskServiceDefaultPort,
+		engineapi.InstanceManagerInstanceServiceDefaultPort,
+	}
+	for _, port := range ports {
+		livenessProbes = append(livenessProbes, fmt.Sprintf("nc -zv localhost %d > /dev/null 2>&1", port))
+	}
+	if types.IsDataEngineV2(dataEngine) {
+		livenessProbes = append(livenessProbes, fmt.Sprintf("nc -zv localhost %d > /dev/null 2>&1", engineapi.InstanceManagerSpdkServiceDefaultPort))
+
+		processProbe := "[ $(ps aux | grep 'spdk_tgt' | grep -v 'grep' | grep -v 'tee' | wc -l) != 0 ]"
+		livenessProbes = append(livenessProbes, processProbe)
+	}
+	return fmt.Sprintf("test $(%s; echo $?) -eq 0", strings.Join(livenessProbes, " && "))
+}
+
 func (imc *InstanceManagerController) createInstanceManagerPodSpec(im *longhorn.InstanceManager, tolerations []corev1.Toleration, registrySecret string, nodeSelector map[string]string, dataEngine longhorn.DataEngineType) (*corev1.Pod, error) {
-	podSpec, err := imc.createGenericManagerPodSpec(im, tolerations, registrySecret, nodeSelector)
+	logDirectory, err := imc.ds.GetSettingValueExisted(types.SettingNameLogDirectory)
```

Comment:
Do we need to:
1. Check if the path format is valid
2. Have a fallback default value if the setting is invalid

Reply:
Addressed. Thank you.